paz/models/detection/frustum_pointnet.py

```diff
@@ -0,0 +1,177 @@
+                         size_class_label_shape=(), size_residual_label_shape=(3,), batch_size=32):
+    end_points = {}
+
+    point_cloud = Input(point_cloud_shape, name="frustum_point_cloud", batch_size=batch_size)
+    one_hot_vec = Input(one_hot_vec_shape, name="one_hot_vec", batch_size=batch_size)
+    mask_label = Input(mask_label_shape, name="seg_label", batch_size=batch_size)
+    center_label = Input(center_label_shape, name="box3d_center", batch_size=batch_size)
+    heading_class_label = Input(heading_class_label_shape, name="angle_class", batch_size=batch_size)
+    heading_residual_label = Input(heading_residual_label_shape, name="angle_residual", batch_size=batch_size)
+    size_class_label = Input(size_class_label_shape, name="size_class", batch_size=batch_size)
+    size_residual_label = Input(size_residual_label_shape, name="size_residual", batch_size=batch_size)
+
+    logits = Instance_Seg_Net(point_cloud, one_hot_vec)  # bs,n,2
+    end_points['mask_logits'] = logits
+
+    # Mask Point Centroid
+    object_point_cloud_xyz, mask_xyz_mean, end_points = point_cloud_masking(point_cloud, logits, end_points)
+
+    # T-Net
+    center_delta = TNet(object_point_cloud_xyz, one_hot_vec)  # (32,3)
+
+    stage1_center = center_delta + mask_xyz_mean  # Bx3
+    end_points['stage1_center'] = stage1_center
+    # Get object point cloud in object coordinate
+    object_point_cloud_xyz_new = object_point_cloud_xyz - tf.expand_dims(center_delta, 1)
+
+    # 3D Box Estimation
+    box_pred = Box_Est_Net(object_point_cloud_xyz_new, one_hot_vec)  # (32, 59)
+
+    end_points = parse_output_to_tensors(box_pred, end_points)
+    end_points['center'] = end_points['center_boxnet'] + stage1_center  # Bx3
+
+    logits = end_points['mask_logits']
+    mask = end_points['mask']
+    stage1_center = end_points['stage1_center']
+    center_boxnet = end_points['center_boxnet']
+    heading_scores = end_points['heading_scores']  # BxNUM_HEADING_BIN
+    heading_residuals_normalized = end_points['heading_residuals_normalized']
+    heading_residuals = end_points['heading_residuals']
+    size_scores = end_points['size_scores']
+    size_residuals_normalized = end_points['size_residuals_normalized']
+    size_residuals = end_points['size_residuals']
+    center = end_points['center']
+
+    logits = Lambda(lambda x: x, name="InsSeg_out")(logits)
+    box3d_center = Lambda(lambda x: x, name="center_out")(center)
+    heading_scores = Lambda(lambda x: x, name="heading_scores")(heading_scores)
+    heading_residual = Lambda(lambda x: x, name="heading_residual")(heading_residuals)
+    heading_residuals_normalized = Lambda(lambda x: x, name="heading_residual_norm")(heading_residuals_normalized)
+    size_scores = Lambda(lambda x: x, name="size_scores")(size_scores)
+    size_residual = Lambda(lambda x: x, name="size_residual")(size_residuals)
+    size_residuals_normalized = Lambda(lambda x: x, name="size_residual_norm")(size_residuals_normalized)
```

Comment:
Lambda layers seem unnecessary. If this is meant to rename the tensors then another way should be found rather than performing a dummy computation.

Reply:
This is done to assign a name to the tensors in the model. Replaced them with tf.identify